### PR TITLE
Harsh/book appointment

### LINKED ISSUE #48: Feature - [User] Book appointment screen
User can book doctor appoint ment 

Screen UI link
https://drive.google.com/file/d/1gy8l0zEG-O7PDAliVfRsWTOfR0I2tthe/view?usp=drive_link

- [x] Display all doctor display here
- [x] User should be able see 15 days date only
- [x] If date or time slot is already booked then it would be disable
- [x] Book appointment button would enable by default 
- [x] Show validation if user has not selected date and time slot
- [x] Show conformation before appointment book
- [x] Once appointment is booked it would be in pending for confirmation state

**Estimation - 20 hours**

### Check List

- [x] I have linked my PR with this task
- [x] I have added comment after completing this task
- [x] I have followed all [guidelines](https://github.com/jaykishan-sewak/doctor-appoint-app/tree/main) for code

### PR DESCRIPTION
**Pull request by HarshPanchalInfusion**
Book Doctor Appointment PR

### CODE REVIEW

File: app/src/main/java/com/android/doctorapp/repository/AppointmentRepository.kt [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            ApiResponse.create(e.fillInStackTrace())
+        }
+    }
+
+    suspend fun getDoctorById(
+        userId: String,
+        fireStore: FirebaseFirestore
+    ): ApiResponse<UserDataRequestModel> {
+        return try {
+            val response = fireStore.collection("user_data")
```
Comment (vikas-infusion) [inline app/src/main/java/com/android/doctorapp/repository/AppointmentRepository.kt]: Please take table name from constant file and also field

File: app/src/main/java/com/android/doctorapp/ui/authentication/login/LoginViewModel.kt [hunk 2]
```diff
@@ -45,10 +45,10 @@ class LoginViewModel @Inject constructor(
 
     var googleSignInClient: GoogleSignInClient
 
-    val email: MutableLiveData<String> = MutableLiveData()
+    val email: MutableLiveData<String> = MutableLiveData("namuneelgar@gmail.com")
```
Comment (vikas-infusion) [inline app/src/main/java/com/android/doctorapp/ui/authentication/login/LoginViewModel.kt]: Remove hard code login credentials

File: app/src/main/java/com/android/doctorapp/util/extension/ContextExtension.kt [hunk 3]
```diff
@@ -95,3 +98,34 @@ fun Context.selectDate(maxDate: Long?, minDate: Long?, handleClick: (date: Strin
...
+        targetDateFormat.format(calendar.time)
+    } catch (e: Exception) {
+        e.printStackTrace()
+        ""
+    }
+}
+
+fun convertTime(originalTimeStr: String): String {
+    return try {
+        val originalTimeFormat = SimpleDateFormat("EEE MMM dd HH:mm:ss z yyyy", Locale.getDefault())
```
Comment (vikas-infusion) [inline app/src/main/java/com/android/doctorapp/util/extension/ContextExtension.kt]: format define in constant file

File: app/src/main/res/drawable/book_appointment_bg.xml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape
+    xmlns:android="http://schemas.android.com/apk/res/android">
+    <padding
+        android:left="2dp"
+        android:top="2dp"
```
Comment (vikas-infusion) [inline app/src/main/res/drawable/book_appointment_bg.xml]: take value from dimens file

File: app/src/main/java/com/android/doctorapp/repository/AppointmentRepository.kt (hunk 1 above)
Comment (HarshPanchalInfusion) [reply app/src/main/java/com/android/doctorapp/repository/AppointmentRepository.kt]: Done

File: app/src/main/java/com/android/doctorapp/util/extension/ContextExtension.kt (hunk 3 above)
Comment (HarshPanchalInfusion) [reply app/src/main/java/com/android/doctorapp/util/extension/ContextExtension.kt]: Done

File: app/src/main/res/drawable/book_appointment_bg.xml (hunk 4 above)
Comment (HarshPanchalInfusion) [reply app/src/main/res/drawable/book_appointment_bg.xml]: Done

File: app/src/main/java/com/android/doctorapp/ui/authentication/login/LoginViewModel.kt (hunk 2 above)
Comment (HarshPanchalInfusion) [reply app/src/main/java/com/android/doctorapp/ui/authentication/login/LoginViewModel.kt]: Done

**Review by vikas-infusion — APPROVED** [review]

**Pull request merged**